pkg/controller/statefulset/stateful_set_control.go

```diff
@@ -591,6 +595,12 @@ func (ssc *defaultStatefulSetControl) updateStatefulSet(
 						replicas[i].Name)
 				}
 				return &status, nil
+			} else if !isAvailable && decreaseAndCheckMaxUnavailable(scaleMaxUnavailable) {
+				if waitTime > 0 {
+					// make sure we check later
+					durationStore.Push(getStatefulSetKey(set), waitTime)
+				}
+				break
```

Comment:
klog.V(4).Infof(
			"StatefulSet %s/%s Pod %s is unavailable, and break pods scale",
			set.Namespace,
			set.Name,
			replicas[i].Name)